### DISCUSSION TITLE
Unit Tests

### DISCUSSION THREAD

**Discussion by DeeEmm** [DeeEmm/DIY-Dyno#4] [Code Development]
I've created a Unit Tests folder with some basic (blank) unit tests for a load cell and tacho. 

A unit test is the simplest and smallest stand alone building block. For example reading data from a sensor to use as a tacho signal.  It is very much like the code examples you see published by vendors such as adafruit - the bare basic code necessary to get the functionality required.

The idea here is that contributors can clone the repository, write some code to get the ESP32 reading data from a load cell or tacho sensor or whatever and then report it to the serial monitor. Then once it is working create a pull request to push the code into the project.

Once I've finished porting the main GUI code I can then utilise these unit tests in the main code.

I figure that this is the simplest way to get contributors involved and allows me to concentrate on the GUI and boiler plate code.

If anyone has any questions, or needs help, please feel free to post in this thread.

/DM

**Comment by patracy** [comment]
I have the chassis dyno pulley and belt setup squared away.  What should I buy next for this?  I also have a SCR for the eddy current brake that uses 0-5v to control it.  The nice thing about this dyno is that it has a 3ph motor to spin the dyno on it's own without a vehicle.   So I can plot the inertia without having any vehicle on it or other variables to have to worry about.

**Reply by DeeEmm** [reply]
I would start to work out what kind of load cell configuration you might need. Take a look at the proposed loads and then work out the load cell specs and torque arm setup. You might also want to consider a VFD to drive the motor if it doesn't already have one

**Reply by patracy** [reply]
There's already a load cell on it, I just need to find out what it's rated at.  And yeah, the 3ph motor has a VFD to drive it.
